src/schedule/dma_multi_chan_domain.c

```diff
@@ -132,7 +132,7 @@ static int dma_multi_chan_domain_register(struct ll_schedule_domain *domain,
 		goto out;
 
 	for (i = 0; i < dma_domain->num_dma; ++i) {
-		for (j = 0; j < dmas[i].plat_data.channels; ++j) {
+		for (j = 0; j < dmas[i].plat_data.channels && (&(dmas[i].chan[j]) != NULL); ++j) {
```

Comment:
I don't think this can reasonably make sense as long as `dmas != NULL` or somewhere below (like if `int *x = (int *)-4` then `&x[1] == NULL`. The address of that array element cannot be NULL.

Reply:
independent of dma[i], the dma[i].chan is allocated in the probe function.
if probe function is not invoked then dma[i].chan will be NULL.

Reply:
then just check `dma[i].chan` once - not `&dma[i].chan[j]`

Reply:
done